### PR TITLE
YD-191 Use standard date/time instead of epoch

### PR DESCRIPTION
**Pull request by Bert-R**
Extended the tests too

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -67,4 +67,18 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 		int num = sequenceNumber++
 		return "$baseTimestamp$num"
 	}
-}
+
+	def assertDateTimeInLimits(dateTimeString, int rangeSeconds = 10, Date comparisonDateTime = null)
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: Possible alternative name is `assertDateEquals` with `rangeSeconds` as third parameter (default 0).

File: appservice/src/intTest/groovy/nu/yona/server/BasicBuddyTest.groovy [hunk 2]
```diff
@@ -80,6 +80,7 @@ class BasicBuddyTest extends AbstractAppServiceIntegrationTest
 		def buddyConnectRequestMessages = response.responseData._embedded.messages.findAll{ it."@type" == "BuddyConnectRequestMessage"}
 		buddyConnectRequestMessages.size() == 1
 		buddyConnectRequestMessages[0].nickname == richard.nickname
+		assertDateTimeInLimits(buddyConnectRequestMessages[0].creationTime)
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/BasicBuddyTest.groovy]: I think `assertDateEquals(buddyConnectRequestMessages[0].creationTime, new Date(), 10)` is more understandable

**Comment by aristotelos** [comment]
Looks good, added a minor suggestion

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy (hunk 1 above)
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: I don't think that's a good idea:
* The ``equals`` part of the name conflicts with the range. If the range is non-zero, then it's a range or limits assertion.
* Making the default zero is a call for trouble for true client/server testing. Currently, the tests run against local host, but we must also be able to run them against a remote host. In that case, clock synchronization will be in place, but not up to the microsecond. That's why I made the range default 10 seconds.

**Comment by Bert-R** [comment]
Replied your suggestions.

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy (hunk 1 above)
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: More frameworks have e.g. `assertEquals(double expected, double actual, double epsilon)`, so I do not agree with the first part. However, it is a good idea to not allow the comparison without the third parameter.

Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: OK, I wasn't aware of that. Will make the change.

**Comment by Bert-R** [comment]
Updated as requested.

**Pull request merged**